docker-compose.yml

```diff
@@ -0,0 +1,211 @@
+version: '3.8'
+
+services:
+  angular-build:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
+    container_name: oppia-angular-build
+    volumes:
+      - .:/app/oppia
+      - shared_node_modules:/app/oppia/node_modules
+      - shared_third_party:/app/oppia/third_party
+      - shared_proto_files:/app/oppia/proto_files
+      - shared_js_ts_proto_files:/app/oppia/extensions/classifiers/proto
+      - shared_assets_dir:/app/oppia/assets
+      - shared_angular_build:/app/oppia/dist
+    healthcheck:
+      test: test "$(ls -A /app/oppia/dist)" || exit 1
+      timeout: 30s
+      interval: 30s
+      retries: 5
+      start_period: 20m
+    # added max-old-space-size to avoid out of memory error, check this link for more info
+    # https://stackoverflow.com/questions/62663167/dockerizing-react-in-production-mode-fatal-error-ineffective-mark-compacts-nea
+    command: >
+      sh -c '
+        trap "rm /app/oppia/node_modules/@angular/compiler-cli/ngcc/__ngcc_lock_file__" EXIT
```

Comment:
Why do we need the `rm` command?

Reply:
to explicitly remove the webpack compiler instance on EXIT, otherwise on restarting the service, it says that there is already a lock file present which webpack refers to check whether the compiler instance is active or not

Reply:
Are we currently doing this through start.py as well?

Reply:
no, but for gracefully stop the container we need this. graceful stopping means deleting the compiler instance which manytimes is not properly done when we stop containers. so we need to have this